### PR TITLE
Add Dockerfile

### LINKED ISSUE #178: Any docker images?
Are there any docker files or pre-build images for this project?

### PR DESCRIPTION
**Pull request by elemoine**
This PR adds a Dockerfile. The built image is based on the [official postgres 9.6 image](https://hub.docker.com/_/postgres/).

The goal of this Docker image is to make it easy for people to install and try out the Pointcloud extension.

Fixes #178.

### CODE REVIEW

**Comment by pblottiere** [comment]
Hi @elemoine!

Thanks for the Dockerfile, I think it's a very good thing to have!

The docker build is OK, but i have an issue when I want to create the pointcloud extension on the Docker container:

```$ createdb mytests
$ psql mytests
psql (9.6.9)
Type "help" for help.

mytests=# create extension postgis;
CREATE EXTENSION
mytests=# create extension pointcloud;
ERROR:  could not open extension control file "/usr/share/postgresql/9.6/extension/pointcloud.control": No such file or directory
```

Actually, the `pointcloud.control` file is there:

```
$ find . -name pointcloud.control
./usr/share/postgresql/10/extension/pointcloud.control
```

**Comment by elemoine** [comment]
Thanks for testing @pblottiere. I'll take a look at the issue. Also, it would be good to compile pointcloud with lazperf support. And we could now use Postgres 10 instead of 9.6.

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        autoconf \
+        automake \
+        zlib1g-dev \
+        postgresql-server-dev-all \
+        libxml2-dev \
+    && rm -rf /var/lib/apt/lists/* \
+    && git clone https://github.com/pgpointcloud/pointcloud \
+    && cd pointcloud \
+    && ./autogen.sh \
+    && ./configure CFLAGS="-Wall -Werror -O2 -g" \
```
Comment (jrobichaud) [inline docker/Dockerfile]: Use the following code instead to fix this https://github.com/pgpointcloud/pointcloud/pull/182#issuecomment-396261740

```suggestion
    && ./configure --with-pgconfig=/usr/lib/postgresql/${POSTGRES_VERSION}/bin/pg_config CFLAGS="-Wall -Werror -O2 -g" \
```

See documentation: https://github.com/pgpointcloud/pointcloud#build

**Comment by pblottiere** [comment]
Superseded by https://github.com/pgpointcloud/pointcloud/pull/265.

**Pull request closed without merging**